WalletWasabi/WabiSabi/Backend/Rounds/Round.cs

```diff
@@ -85,12 +86,36 @@ public Round(Round blameOf) : this(blameOf.RoundParameters)
 				var weightCredentialResponse = WeightCredentialIssuer.HandleRequest(zeroWeightCredentialRequests);
 
 				alice.SetDeadlineRelativeTo(ConnectionConfirmationTimeout);
+				foreach (var op in alice.Coins.Select(x => x.Outpoint))
+				{
+					if (RemoveAlicesNoLock(x => x.Coins.Select(x => x.Outpoint).Contains(op)) > 0)
+					{
+						Logger.LogInfo("Updated Alice registration.");
+					}
+				}
 				Alices.Add(alice);
 
 				return new(alice.Id, amountCredentialResponse, weightCredentialResponse);
 			}
 		}
 
+		public int RemoveAlices(Predicate<Alice> match)
+		{
+			lock (Lock)
+			{
+				return RemoveAlicesNoLock(match);
+			}
+		}
+
+		private int RemoveAlicesNoLock(Predicate<Alice> match)
```

Comment:
I use this SomethingNoLock pattern all the time, but I'm not very happy with this. Not sure how to do it better.